Simplenote/ExcerptMaker.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+
+// MARK: - ExcerptMaker: Generate excerpt from a note with specified keywords
+//
+@objc
+final class ExcerptMaker: NSObject {
+    private var regexp: NSRegularExpression?
+    private var keywords: [String]?
+
+    /// Update with search keywords
+    ///
+    @objc
+    func update(withKeywords keywords: [String]?) {
+        guard keywords != self.keywords else {
+            return
+        }
+
+        self.keywords = keywords
+
+        guard let keywords = keywords, !keywords.isEmpty else {
+            regexp = nil
+            return
+        }
+
+        regexp = NSRegularExpression.regexForExcerpt(withKeywords: keywords)
```

Comment:
Sugestion! You can decouple / split things a bit:

```
private var keywords: [String]? {
   didSet {
      guard oldValue != keywords else {
         return
      }
      refreshRegex()
   }
}
```

Reply:
Good point!